stl/inc/mdspan

```diff
@@ -125,13 +125,11 @@ public:
     using size_type  = make_unsigned_t<index_type>;
     using rank_type  = size_t;
 
-    // TRANSITION: doesn't account for extended integer types
-    static_assert(_Is_any_of_v<remove_cv_t<_IndexType>, signed char, unsigned char, short, unsigned short, int,
-                      unsigned int, long, unsigned long, long long, unsigned long long>,
-        "N4928 [mdspan.extents.overview]/2 "
-        "requires that extents::index_type be a signed or unsigned integer type.");
-
-    static_assert(((_Extents == dynamic_extent || _Extents <= (numeric_limits<_IndexType>::max)()) && ...));
+    static_assert(_Is_standard_integer<_IndexType>,

```

Comment:
This removes the `TRANSITION` comment, but still doesn't check for extended integer types. "[un]signed integer type" is the union of "standard [un]signed integer type" and "extended [un]signed integer type". There's also a test that instantiates all the allowed types, so it should have `_Signed128` and `_Unsigned128` cases added. I'd probably also want to try out an `mdspan` with the extended types, to make sure it and the corresponding layout mappings are also working.

Reply:
Both `_Signed128` and `_Unsigned128` are not *extended-integer types* (since they don't model `integral`, see [[basic.fundamental]/1](http://eel.is/c++draft/basic.fundamental#1)), but *integer-class types* (see [[iterator.concept.winc]/4](http://eel.is/c++draft/iterator.concept.winc#4)). And since MSVC-STL does not define any *extended-integer types*, I think this is the best thing that could be done.

I think we can support *integer-class types* as an extension, but I would like to hear maintainer's opinion on this topic.

Reply:
See #3457.

Reply:
Supporting integer-class types as an extension seems like it could require far-reaching changes to constraints etc. in what is already going to be a very large and complicated feature. I would prefer not to overreach.

Reply:
> Both `_Signed12`8 and `_Unsigned128` are not extended-integer types (since they don't model integral, see [[basic.fundamental]/1](http://eel.is/c++draft/basic.fundamental#1)), but integer-class types.

Oh, I see, thank you! Yet another category of integer types to remember...